pkg/utils/jsonutils.go

```diff
@@ -58,3 +60,29 @@ func FlattenSingleValue(key string, m map[string]interface{}, child interface{})
 		m[key] = child
 	}
 }
+
+// GetString extracts a string value from JSON data based on the provided keys.
+// It uses the supplied workBuf to avoid unnecessary allocations and resizing.
+// If the value is not a string or if an error occurs, it returns an appropriate error.
+func GetString(data []byte, workBuf []byte, keys ...string) ([]byte, error) {
```

Comment:
nit: rename this to be `GetStringFromJson`

Reply:
updated